core/src/main/java/org/apache/iceberg/TableScanContext.java

```diff
@@ -168,7 +172,17 @@ Long fromSnapshotId() {
   TableScanContext fromSnapshotId(long id) {
     return new TableScanContext(snapshotId, rowFilter, ignoreResiduals,
         caseSensitive, colStats, projectedSchema, selectedColumns, options, id, toSnapshotId,
-        planExecutor);
+        planExecutor, fromSnapshotInclusive);
+  }
+
+  TableScanContext fromSnapshotIdInclusive(long id) {
```

Comment:
Shall we rename the existing methods (`fromSnapshotId`) to indicate they configure the start snapshot exclusive?
We may also set `fromSnapshotInclusive` explicitly in existing methods.

Reply:
agree. since this is a non-public class, it is safe to rename the old setter of `fromSnapshotId` to `fromSnapshotIdExclusive`